src/csharp/Grpc.Core.Tests/CallCancellationTest.cs

```diff
@@ -189,5 +189,41 @@ public void CanDisposeDefaultCancellationRegistration()
 
             using (obj) {}
         }
+
+        [Test]
+        public async Task ServerStreamingCall_CancelAfterServerHasFinished()
+        {
+            // Test to verify the fix for https://github.com/grpc/grpc/issues/17761 that was
+            // fixed in c-core https://github.com/grpc/grpc/issues/18234
+            // Scenario: the client cancels the request after the server has already successfully
+            // returned the results but before the client starts reading.
+            // Expected behaviour is that the client sees a Cancelled status.
```

Comment:
expected behavior is that we see Cancelled status AND no messages are received? (at least it looks like that's what you're asserting).

Reply:
Updated the comments clarify the expected behaviour